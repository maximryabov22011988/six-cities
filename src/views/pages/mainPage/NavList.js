import React from 'react';
import PropTypes from 'prop-types';

import withActiveItem from '../../hocs/withActiveItem';
import NavItem from './navList/NavItem';

const propTypes = {
  currentCity: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.arrayOf(PropTypes.number),
      zoom: PropTypes.number
    })
  ).isRequired,
  onChangeCity: PropTypes.func.isRequired
};

function NavList({ currentCity, cities, onChangeCity }) {
  return (
    <ul className="locations__list tabs__list">
      {cities.map(city => {
        const WrappedNavItem = withActiveItem(NavItem);
        return (
          <WrappedNavItem
            key={city.name}
            className="tabs__item"
            city={city}
            isActive={city.name === currentCity}
            onChangeCity={onChangeCity}
          />
        );
      })}
    </ul>
  );
}

NavList.propTypes = propTypes;

export default NavList;
