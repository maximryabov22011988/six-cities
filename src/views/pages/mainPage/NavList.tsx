import * as React from 'react';

import NavItem from './navList/NavItem';

import withActiveItem from '../../hocs/withActiveItem';

import { CurrentCity, onChangeCity } from '../../interfaces';

interface Props {
  cities: Array<CurrentCity>,
  currentCity: string,
  onChangeCity: onChangeCity,
}

function NavList({ currentCity, cities, onChangeCity }: Props) {
  return (
    <ul className="locations__list tabs__list">
      {cities.map((city) => {
        const WrappedNavItem = withActiveItem(NavItem);
        return (
          <WrappedNavItem
            city={city}
            className="tabs__item"
            isActive={city.name === currentCity}
            key={city.name}
            onChangeCity={onChangeCity}
          />
        );
      })}
    </ul>
  );
}

export default NavList;
