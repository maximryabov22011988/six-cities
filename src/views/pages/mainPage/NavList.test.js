import React from 'react';
import renderer from 'react-test-renderer';

import NavList from './NavList';

const offersMock = [
  {
    id: 1,
    city: 'Paris',
    coords: {
      latitude: 52.38333,
      longitude: 4.9
    }
  }
];

it('NavList correctly render', () => {
  const handleChangeCity = jest.fn();
  const navList = renderer
    .create(
      <NavList
        currentCity="Paris"
        cities={offersMock}
        onChangeCity={handleChangeCity}
      />
    )
    .toJSON();

  expect(navList).toMatchSnapshot();
});
