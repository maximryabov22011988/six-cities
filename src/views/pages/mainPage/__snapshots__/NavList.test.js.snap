// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList correctly render 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="tabs__item tabs__item--active locations__item-link"
      href="#"
    >
      <span>
        Paris
      </span>
    </a>
  </li>
</ul>
`;
